toolchain/driver/driver.cpp

```diff
@@ -163,6 +165,22 @@ auto Driver::RunDumpSubcommand(DiagnosticConsumer& consumer,
     args = args.drop_front();
   }
 
+  llvm::StringRef output_file;
+  if (dump_mode == DumpMode::ObjectCode) {
+    if (!args.empty() && args.front().starts_with("--target_triple=")) {
+      target_triple = args.front().split("=").second;
+      args = args.drop_front();
+    }
+    if (!args.empty() && args.front().starts_with("--output_file=")) {
```

Comment:
It'd be preferable to support flags appearing in any order. Maybe we could put a loop around this flag-parsing code? We could keep going until we run out of args or find an argument that doesn't start with `--`, and diagnose any `--` flag that we don't recognize.